arborio/jsonio.cpp

```diff
@@ -0,0 +1,328 @@
+#include <fstream>
+#include <iomanip>
+#include <map>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "arbor/util/any_visitor.hpp"
+#include "arborio/jsonio.hpp"
+#include "json_helpers.hpp"
+
+namespace arborio {
+
+jsonio_error::jsonio_error(const std::string& msg):
+    arbor_exception(msg) {}
+
+jsonio_unused_input::jsonio_unused_input(const std::string& key):
+    jsonio_error("Unused input parameter: \"" + key + "\"")
+{}
+
+jsonio_decor_global_load_error::jsonio_decor_global_load_error(const std::string err):
+    jsonio_error("Decor: error loading global parameters: " + err)
+{}
+
+jsonio_decor_global_set_error::jsonio_decor_global_set_error(const std::string err):
+    jsonio_error("Decor: error setting global parameters: " + err)
+{}
+
+jsonio_decor_local_missing_region::jsonio_decor_local_missing_region():
+    jsonio_error("Decor: local parameters must include region label")
+{}
+
+jsonio_decor_local_revpot_mech::jsonio_decor_local_revpot_mech():
+    jsonio_error("Decor: cannot implement local reversal potential methods")
+{}
+
+jsonio_decor_local_load_error::jsonio_decor_local_load_error(const std::string err):
+    jsonio_error("Decor: error loading local parameters: " + err)
+{}
+
+jsonio_decor_local_set_error::jsonio_decor_local_set_error(const std::string err):
+    jsonio_error("Decor: error painting local parameters: " + err)
+{}
+
+jsonio_decor_mech_missing_region::jsonio_decor_mech_missing_region():
+    jsonio_error("Decor: mechanism description does not include region label")
+{}
+
+jsonio_decor_mech_missing_name::jsonio_decor_mech_missing_name():
+    jsonio_error("Decor: mechanism description does not include mechanism name")
+{}
+
+jsonio_decor_mech_set_error::jsonio_decor_mech_set_error(const std::string& reg, const std::string& mech, const std::string& err):
+    jsonio_error("Decor: Error painting mechanism \"" + mech + "\" on region \"" + reg + "\": " + err)
+{}
+
+jsonio_json_parse_error::jsonio_json_parse_error(const std::string err):
+    jsonio_error("Error parsing JSON : " + err)
+{}
+
+arb::cable_cell_parameter_set load_cable_cell_parameter_set(nlohmann::json& params_json) {
```

Comment:
I am wondering if this here

https://github.com/nlohmann/json#basic-usage

is not a better option?

Reply:
I'll check it out, thanks!